CHANGELOG.md

```diff
@@ -8,6 +8,7 @@
 
 ### Enhancements
 * Add support for the `GUID` data type. It can be used as primary key and is indexable. (PR [#2120](https://github.com/realm/realm-dotnet/pull/2120))
+* Replaced the implementation of the string query parser (the one used for realm.All().Filter("some-string-query")). This results in ~5% reduction of the size of the native binary while keeping the query execution times on par with the old parser.
```

Comment:
```suggestion
* Replaced the implementation of the string query parser (the one used for [`realm.All().Filter("some-string-query")`](https://docs.mongodb.com/realm-sdks/dotnet/10.0.0-beta.3/reference/Realms.CollectionExtensions.html#Realms_CollectionExtensions_Filter__1_System_Linq_IQueryable___0__System_String_)). This results in ~5% reduction of the size of the native binary while keeping the query execution times on par with the old parser. (PR [#2185](https://github.com/realm/realm-dotnet/pull/2185), Core upgrade)
```

At the end of the change entry we try to add some additional information, depending on the change type:
* If there's an issue filed for the change (typically a bug or a feature request), we add (Issue [#123](https://github.com/realm/realm-dotnet/issue/123)). This allows the person who filed the issue and everyone interested to see that the new release includes what they were asking for.
* If there's no issue we add a link to the PR that introduced the change - e.g. (PR [#2120](https://github.com/realm/realm-dotnet/pull/2120)) like above. This allows us to quickly find the context of a change by looking at the changelog, which is (typically) more readable than the git history.
* Finally, if all a PR does is to bump the core version and we get some fixes for free, we add (Core upgrade). 

In this case, I proposed a combination of 2 and 3 because the filesize reduction is thanks to the Core upgrade, but the PR includes some changes of its own, so worth linking to it.

As for the link to the API docs - we typically don't bother linking to the API from the changelog, but in this case, it concerns a feature that may not be super heavily used (as I'm fairly certain most devs will prefer to use LINQ), so adding a link makes it more prominent and will allow people who are curious about it to find more info quickly.